lib/shared/addon/utils/util.js

```diff
@@ -345,6 +345,52 @@ export function isBadTld(name) {
   }
 }
 
+export function removeEmpty(obj){
+  return Object.keys(obj)
+    .filter(k => obj[k] !== null && obj[k] !== undefined && (typeof obj[k] !== 'object' || (Object.keys(obj[k]).filter((key) => key !== 'type').length)))
```

Comment:
This is hard to read and reason about. Can we simplify this? You can use the `isEmpty` helper for the first checks. Which would simplify it a bit. I think abstracting to another function that returns a bool for the second `or` condition would make it easier to reason about as well. 

```  
return Object.keys(obj)
    .filter(k => !isEmpty(obj[k]) && (typeof obj[k] !== 'object' || keyNotType(obj[k])))
    .reduce((newObj, k) =>
      typeof obj[k] === 'object' ?
        Object.assign(newObj, {[k]: removeEmpty(obj[k])}) :
        Object.assign(newObj, {[k]: obj[k]}),
      {});

  function keyNotType(k) {
    return Object.keys(k).filter(key => key !== 'type').length > 0;
  }
``` 
for example. Naming is up to you.